js/app/application.js

```diff
@@ -142,12 +144,13 @@ const Application = new Knowledge.Class({
         if (engine instanceof MoltresEngine.MoltresEngine)
             return;
 
+        this._check_for_update();
         try {
-            engine.update_and_preload_default_domain();
+            engine.preload_default_domain_shards();
         } catch (e if e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.NOT_FOUND)) {
             // No content. If updates are pending then show a nice dialog
-            let subs = engine.get_default_domain().get_subscription_entries();
-            if (subs.some(entry => !entry.disable_updates)) {
+            let entry = engine.get_default_domain().get_subscription_entry();
+            if (!entry.disable_updates) {
```

Comment:
How about I rewrite the update function to use `get_subscription_entries` when I move it to application (and update multiple subscriptions if there were any). Then we can keep `get_subscription_entries` as the exposed API from domain and extend it to handle such internally down the line.